src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/Deflater.cs

```diff
@@ -134,9 +134,12 @@ internal int GetDeflateOutput(byte[] outputBuffer)
 
         private unsafe ZErrorCode ReadDeflateOutput(byte[] outputBuffer, ZFlushCode flushCode, out int bytesRead)
         {
+            Debug.Assert(null != outputBuffer, "Can't pass in a null output buffer!");
```

Comment:
I realize we don't typically use yoda conditionals, but I just copy/pasted some existing asserts from some of the callers of this method.